File: ex5/ex5.erl
-module(ex5).
-import(lists, [member/2]).
-export([interseccao/2]).

interseccao([], _) -> [];
interseccao([Head_a|A], B) ->
  case
    {lists:member(Head_a, B)} of
      {true} -> [Head_a | interseccao(A, B)];
      {false} -> interseccao(A, B)
  end.

% c(ex5).
% ex5:interseccao([3, 6, 5, 7], [9, 7, 5, 1, 3]).

File: ex5 (Victor)/ex5.hs
interseccao :: [Integer] -> [Integer] -> [Integer]
interseccao [] _ = []
interseccao (head_a:a) b | head_a `elem` b = head_a : interseccao a b
                         | otherwise = interseccao a b

-- ghci ex5.hs
main = do 
   putStrLn "The intersection of two lists is:"  
   print(interseccao [3, 6, 5, 7] [9, 7, 5, 1, 3])

-- head_a         a                  b
-- 3          [6, 5, 7]       [9, 7, 5, 1, 3]

-- interseccao [3, 6, 5, 7] [9, 7, 5, 1, 3]
-- 3 : interseccao [6, 5, 7] [9, 7, 5, 1, 3]
-- 3 : interseccao [5, 7] [9, 7, 5, 1, 3]
-- 3 : 5 : interseccao [7] [9, 7, 5, 1, 3]
-- 3 : 5 : 7 : interseccao [] [9, 7, 5, 1, 3]
-- 3 : 5 : 7 : []